### ISSUE TITLE
Evaluation scores dont match inference results 

### ISSUE THREAD

**Issue by LenaMetzen** [open-edge-platform/anomalib#1548]
### Describe the bug

I trained a GANomaly model and got good scores for the evaluation metrics 
- image_F1Score (0,79) and image_AUROC (0,88).
- Dataset contains 750 normal and 130 abnormal images

When running inference all images (normal and abnormal) are predicted as anomalous. 
- Either running inference on unseen normal and abnormal images
- Or on already seen normal training images
- Not a single image is predicted as normal

How can I get good scores but bad prediction results?
- Due to the bad prediction results on the already seen training data overfitting cant be the reason

### Dataset

Other (please specify in the text field below)

### Model

GANomaly

### Steps to reproduce the behavior

1. Install anomalib using git clone, create conda environment, install all dependencies, install pytorch for cuda...
2. Train a GANomaly model ona a custom dataset with adjusted hyperparameters
3. Receiving good results on the evaluation set
4. Running inference on new and already seen images

### OS information

OS information:
- OS: [Windows 10]
- Python version: [3.10]
- Anomalib version: [0.7.0]
- Torch version: [2.1.1]

### Expected behavior

Prediction results should match the Evaluation Metrics

### Screenshots

**Evaluation scores**

![Bild1](https://github.com/openvinotoolkit/anomalib/assets/101172221/9bce63c9-f459-4a93-82ad-2267f0822b35)

**Prediction results for new (unseen) images**

<img width="300" alt="Bild4" src="https://github.com/openvinotoolkit/anomalib/assets/101172221/4988d47f-ed8e-4fc9-9c8f-a5413fcd97e6">

**Prediction results for normal training images (already seen images)**
<img width="413" alt="Bild3" src="https://github.com/openvinotoolkit/anomalib/assets/101172221/98d71b75-1ac8-4e4e-a9cc-b7d65d1baeb0">

**Train loss over epochs**
<img width="195" alt="Bild5" src="https://github.com/openvinotoolkit/anomalib/assets/101172221/05b8d572-ed9f-42da-bfff-7e12c8405ee2">

**Image F1 Score**
<img width="198" alt="Bild6" src="https://github.com/openvinotoolkit/anomalib/assets/101172221/3d639e6d-d58a-4d25-906d-dbc3628f4c12">

### Pip/GitHub

GitHub

### What version/branch did you use?

0.7.0 (latest)

### Configuration YAML

```yaml
dataset:
  abnormal_dir: NIO_sortierung_nach_DIN
  eval_batch_size: 32
  extensions: null
  format: folder
  image_size: 256
  mask_dir: null
  name: SolderJoint
  normal_dir: IO_sortierung_nach_DIN
  normal_test_dir: null
  normalization: imagenet
  num_workers: 0
  root: ./datasets/Pins/gims_download_new/nach_DIN/
  seed: 0
  split_ratio: 0.2
  task: classification
  test_split_mode: from_dir
  train_batch_size: 32
  transform_config:
    eval: null
    train: null
  val_split_mode: from_test
  val_split_ratio: 0.5
logging:
  log_graph: false
  logger:
  - tensorboard
metrics:
  image:
  - F1Score
  - AUROC
  threshold:
    manual_image: null
    method: adaptive
model:
  add_final_conv: true
  beta1: 0.5
  beta2: 0.999
  early_stopping:
    metric: image_F1Score
    mode: max
    patience: 8
  extra_layers: 1
  latent_vec_size: 100
  lr: 0.0002
  n_features: 128
  name: ganomaly
  normalization_method: none
  wadv: 1
  wcon: 50
  wenc: 1
optimization:
  export_mode: null
project:
  path: ./results/ganomaly_Pins_nach_DIN/test
  seed: 42
trainer:
  accelerator: gpu
  accumulate_grad_batches: 1
  auto_lr_find: false
  auto_scale_batch_size: false
  benchmark: false
  check_val_every_n_epoch: 5
  default_root_dir: null
  detect_anomaly: false
  deterministic: false
  devices: 1
  enable_checkpointing: true
  enable_model_summary: true
  enable_progress_bar: true
  fast_dev_run: false
  gradient_clip_algorithm: norm
  gradient_clip_val: 0
  limit_predict_batches: 1.0
  limit_test_batches: 1.0
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 2
  max_epochs: 100
  max_steps: -1
  max_time: null
  min_steps: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
  num_nodes: 1
  num_sanity_val_steps: 0
  overfit_batches: 0.0
  plugins: null
  precision: 32
  profiler: null
  reload_dataloaders_every_n_epochs: 0
  replace_sampler_ddp: true
  strategy: null
  sync_batchnorm: false
  track_grad_norm: -1
  val_check_interval: 1.0
visualization:
  image_save_path: null
  log_images: true
  mode: full
  save_images: true
  show_images: false
```

### Logs

```shell
See screenshots
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 1999swallowthemoon** [comment]
I also encountered this problem in anomalib=2.0.0. How did you solve it at that time? Thank you very much for your answer.

**Comment by samet-akcay** [comment]
Duplicate of #2747. We are aware of this issue, and will propose a fix asap
